syntax = "proto3";

package payment;

service PaymentService {
  rpc ProcessPayment (PaymentRequest) returns (PaymentResponse);
}

message PaymentRequest {
  string provider = 1;
  double amount = 2;
  string currency = 3;
  string customer_id = 4;
}

message PaymentResponse {
  string status = 1;
  string transaction_id = 2;
}
